### PR TITLE
Add multiplayer room music dimming option

### PR DESCRIPTION
**Pull request by OliBomby**
Closes #13173

- Added settings slider "Music (multiplayer room)" (default value is 100%).
- Added a BackgroundTrackManager for handling muting/dimming of the music track.
- The multiplayer room screen now dims the music track by the amount defined in the settings.
- Updated PreviewTrackManager to mute the music track using BackgroundTrackManager and added a 100ms transition.

### CODE REVIEW

File: osu.Game/Screens/OnlinePlay/Multiplayer/MultiplayerMatchSubScreen.cs [hunk 1]
```diff
@@ -274,6 +283,40 @@ protected override void LoadComplete()
...
+
+            base.OnEntering(last);
+        }
+
+        public override bool OnExiting(IScreen next)
+        {
+            volumeMultiplayerRoom.UnbindEvents();
+            backgroundTrackManager.RemoveDimming(1000, Easing.OutQuint);
+
+            return base.OnExiting(next);
```
Comment (ekrctb) [inline osu.Game/Screens/OnlinePlay/Multiplayer/MultiplayerMatchSubScreen.cs:319]: Factor out code duplication.

File: osu.Game/Audio/PreviewTrackManager.cs [hunk 2]
```diff
@@ -23,6 +23,9 @@ public class PreviewTrackManager : Component
         [Resolved]
```
Comment (ekrctb) [inline osu.Game/Audio/PreviewTrackManager.cs:23]: `muteBindable` field can be removed.

**Review by ekrctb — CHANGES_REQUESTED** [review]
I left only nits. Looks like a nice implementation.
Not sure what is going on on the commit history (duplicated commits?)

**Comment by bdach** [comment]
Just so this doesn't go by unsaid, I still disagree with having this as a separate setting, and the descriptions are now actively confusing. There are:
* Master
* Master (window inactive)
* Effect
* Music
* Music (multiplayer room)

I think it's quite messy and nigh indecipherable. And it also runs the risk of someone saying that we need another 3 sliders for another 3 screens. I still think it should really either happen on its own always without a setting (opinionated choice) or be a part of the master (inactive) setting, and the user can potentially decide where the dim is applied.

**Comment by peppy** [comment]
Agree, undecided on which of the proposed directions is better.

Curious why this PR was made before discussion had reached any kind of consensus. I haven't read any of the code and don't intend to until the proposed behaviour is unanimously agreed on.

**Comment by OliBomby** [comment]
I got a bit cocky and made a PR too early. I wrongly thought the discussion reached enough of a consensus to make a PR. Should I close this PR and come back when a consensus is reached or can I leave it here?

What bdach said about the risk of needing another 3 sliders for another 3 screens is right, so I changed my mind. I think it's best if there is a way to configure volume dimming for each screen individually but without using the settings overlay (maybe the volume overlay is good for this?). 

Using the inactive dim setting seems like a worse option, because how are you gonna explain that it controls the master volume dimming for inactive window and controls the music volume for X other screens. That makes everything much more complicated imo.

**Comment by smoogipoo** [comment]
Closing as there's still a lot of differing opinions, direction is not clear yet, and this is a low priority issue for us.

**Pull request closed without merging**